docs/preview/features/message-pumps/service-bus-azure-functions.md

```diff
@@ -0,0 +1,150 @@
+        CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Processing shipment {ShipmentId} for order #{OrderId}", shipment.Id, shipment.Order.Id);
+    }
+}
+```
+
+Processing order messages:
+
+```csharp
+public class OrderServiceBusMessageHandler : IAzureServiceBusMessageHandler<Order>
+{
+    private readonly ILogger _logger;
+
+    public OrderServiceBusMessageHandler(ILogger<OrderServiceBusMessageHandler> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task ProcessMessageAsync(
+        Order order,
+        AzureServiceBusMessageContext messageContext,
+        MessageCorrelationInfo correlationInfo,
+        CancellationToken cancellationToken)
+    {
+        log.LogInformation("Processing order {OrderId} for {OrderAmount} units of {OrderArticle} bought by {CustomerFirstName} {CustomerLastName}", order.Id, order.Amount, order.ArticleNumber, order.Customer.
+    }
+}
+```
+
+Registering both message handlers:
+
+```csharp
+[assembly: FunctionsStartup(typeof(Startup))]
+namespace MessageProcessing
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.Services.WithServiceBusMessageHandler<OrderServiceBusMessageHandler, Order>(messageContext => messageContext.UserProperties["MessageType"] == MessageType.Order)
+                            .WithServiceBusMessageHandler<ShipmentServiceBusMessageHandler, Shipment>(messageContext => messageContext.UserProperties["MessageType"] == MessageType.Shipment);
+        }
+    }
+}
+```
+
+## Activating message routing
+To actually use these registered message handles in the Azure Function, you'll have to register the Azure Service Bus message router.
+This can easily done with the `.AddServiceBusRouting` extension:
+
+```csharp
+[assembly: FunctionsStartup(typeof(Startup))]
+namespace MessageProcessing
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.Services.AddServiceBusMessageRouting()
+                            .WithServiceBusMessageHandler<OrderServiceBusMessageHandler, Order>(messageContext => messageContext.UserProperties["MessageType"] == MessageType.Order)
+                            .WithServiceBusMessageHandler<ShipmentServiceBusMessageHandler, Shipment>(messageContext => messageContext.UserProperties["MessageType"] == MessageType.Shipment);
+        }
+    }
+}
+```
+
+This extension will register an `IAzureServiceBusMessageRouter` interface allows you access to message handling with specific Service Bus operations during the message processing (like dead lettering and abandonning).
+It also registers an more general `IMessageRouter` you can use if the general message routing (with the message raw message body as `string` as incoming message) will suffice.
+
+In any case, this router can be injected in your function:
```

Comment:
We can now inject the message router in our Azure Function and process all messages with it.

This will determine what the matching message handler is and process it accordingly:

Reply:
Updated that line with your suggestion, thanks! Resolving.